File: planix-frontend/src/types/task.ts
export type TaskStatus =
  | "PENDING"
  | "IN_PROGRESS"
  | "COMPLETED"
  | "CANCELLED";

export interface Task {
  id: number;
  title: string;
  description?: string;
  deadline?: string;
  status: TaskStatus;
  completedAt?: string;
  subjectId?: number | null;
}

export interface TaskRequest {
  title: string;
  description?: string;
  deadline?: string;
  subjectId?: number | null;
  status: TaskStatus;
}

File: planix/src/main/java/com/planix/backend/entity/Task.java
package com.planix.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Сущность задачи.
 * Может быть связана с предметом или быть личной.
 */
@Entity
@Table(name = "task")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = {"user", "subject"})
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private LocalDate deadline;

    @Column(nullable = false)
    private String status; //PENDING / IN_PROGRESS / COMPLETED / CANCELLED

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "subject_id")
    private Subject subject;

}